evm/src/executor/backend/mod.rs

```diff
@@ -1141,6 +1150,81 @@ impl DatabaseExt for Backend {
         Ok(())
     }
 
+    fn transact_from_tx(
```

Comment:
now that we have two functions for transact, could we add some comments on what they do and why the name?